app/search/client/provider/result.js

```diff
@@ -79,7 +79,7 @@ Template.DefaultSearchResultTemplate.onCreated(function () {
 
 	// global search
 	this.globalSearchEnabled = this.data.settings.GlobalSearchEnabled;
-	this.data.parentPayload.searchAll = this.globalSearchEnabled;
+	this.data.parentPayload.searchAll = (!!this.globalSearchEnabled && !!this.data.parentPayload.searchAll);
```

Comment:
```suggestion
	this.data.parentPayload.searchAll = false;
```
You can just set it to false